migrator/src/main/scala/com/scylladb/migrator/readers/DynamoDB.scala

```diff
@@ -22,23 +24,32 @@ object DynamoDB {
       if (readThroughput.isEmpty || writeThroughput.isEmpty) Some("25")
       else None
 
-    val df = spark.read
-      .options(
-        (List(
-          source.region.map("region" -> _),
-          source.endpoint.map(ep => "endpoint"                     -> ep.renderEndpoint),
-          source.scanSegments.map(segs => "readPartitions"         -> segs.toString),
-          source.throughputReadPercent.map(pct => "targetCapacity" -> pct.toString),
-          source.maxMapTasks.map(tasks => "defaultParallelism"     -> tasks.toString),
-          throughput.map("throughput" -> _)
-        ).flatten ++ source.credentials.toList.flatMap {
-          case AWSCredentials(accessKey, secretKey) =>
-            List("accesskey" -> accessKey, "secretkey" -> secretKey)
-        }).toMap
-      )
-      .dynamodb(source.table)
+    val jobConf = new JobConf(spark.sparkContext.hadoopConfiguration)
+    def setOptionalConf(name: String, maybeValue: Option[String]): Unit =
+      for (value <- maybeValue) {
+        jobConf.set(name, value)
+      }
+    jobConf.set(DynamoDBConstants.INPUT_TABLE_NAME, source.table)
```

Comment:
why couldn't all below be set as this?

Reply:
if this is about Option parsing then it's not how it's done
you need to set the default value if option passed is None

Reply:
ok, I see, you basically use foreach on collection
getOrElse wouldn't work?

Reply:
Many configuration values are optional in our configuration. They are modeled as `Option[…]` values. The method `setOptionalConf` sets a configuration value if the user provided it, otherwise it does nothing. This is equivalent to what was done here: https://github.com/scylladb/scylla-migrator/pull/5/files#diff-4a83461826fff740a26ca2d8d24aa7c554743398d5a5c215a92c4041d263df13R120

Reply:
> getOrElse wouldn't work?

There is no default value. I set the config only if the user provided a value. Otherwise I do nothing.

Reply:
I think this is more idiomatic than `if (maybeValue.nonEmpty) { … maybeValue.get }`.

I can also replace the `for` with the following, if that reads better:

~~~ scala
maybeValue match {
  case Some(value) => jobConf.set(name, value)
  case None => // Do nothing
}
~~~